invenio_db/ext.py

```diff
@@ -66,8 +68,35 @@ def pathify(base_entry):
         app.extensions["invenio-db"] = self
         app.cli.add_command(db_cmd)
 
+    def build_db_uri(self, app):
+        """Return the connection string if configured or build it from its parts.
+
+        Priority order:
+        1. ``SQLALCHEMY_DATABASE_URI``
+        2. ``DB_{USER,PASSWORD,HOST,PORT,NAME,PROTOCOL}`` configuration variables
+        3. Default URI
+
+        Note: For the ``INVENIO_DB_*`` settings to take effect (i.e. option 2),
+        ``SQLALCHEMY_DATABASE_URI`` needs to be unset.
+        """
+        if uri := app.config.get("SQLALCHEMY_DATABASE_URI", None):
+            return uri
+
+        params = {}
+        for config_name in ["user", "password", "host", "port", "name", "protocol"]:
+            params[config_name] = app.config.get(f"DB_{config_name.upper()}", None)
+
+        if all(params.values()):
+            uri = f"{params['protocol']}://{params['user']}:{params['password']}@{params['host']}:{params['port']}/{params['name']}"
```

Comment:
nit: I felt a bit uneasy looking at this and came across https://docs.sqlalchemy.org/en/20/core/engines.html#creating-urls-programmatically which might also handle some validation :)

Not a blocker, so feel free to take as out of scope and shelve.

Reply:
i've updated the PR to use that function, feels better to not code connection strings by hand

it seems like the protocol is the only actually required parameter for that function, should we relax the `if all(params.values())` check accordingly?
personally i'm slightly leaning towards leaving it as is; i don't think laxness here will buy us a lot, and being stricter makes the opt-in stronger.